tests/unit/admin/test-class-wc-rest-payments-reader-charges-controller.php

```diff
@@ -116,4 +129,272 @@ public function test_get_summary() {
+		$request->set_param( 'payment_id', 42 );
+
+		$response = $this->controller->generate_print_receipt( $request );
+
+		$this->assertInstanceOf( 'WP_Error', $response );
+		$data = $response->get_error_data();
+		$this->assertArrayHasKey( 'status', $data );
+		$this->assertEquals( 500, $data['status'] );
+	}
+
+	public function test_generate_print_receipt_handle_settings_exception() {
+		$order = WC_Helper_Order::create_order();
+
+		$payment_intent = $this->mock_payment_intent();
+
+		$charge = $this->mock_charge( $order->get_id() );
+
+		$settings = $this->mock_settings();
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_intent' )
+			->with( '42' )
+			->willReturn( $payment_intent );
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_charge' )
+			->with( $payment_intent->get_charge_id() )
+			->willReturn( $charge );
+
+		$this->mock_wcpay_gateway
+			->expects( $this->exactly( 1 ) )
+			->method( 'get_option' )
+			->willThrowException( new Exception( 'Something bad' ) );
+
+		$this->mock_receipts_service
+			->expects( $this->never() )
+			->method( 'get_receipt_markup' );
+
+		$request = new WP_REST_Request( 'GET' );
+		$request->set_param( 'payment_id', 42 );
+
+		$response = $this->controller->generate_print_receipt( $request );
+
+		$this->assertInstanceOf( 'WP_Error', $response );
+		$data = $response->get_error_data();
+		$this->assertArrayHasKey( 'status', $data );
+		$this->assertEquals( 500, $data['status'] );
+	}
+
+	public function test_generate_print_receipt_handle_receipt_service_exception() {
+		$order = WC_Helper_Order::create_order();
+
+		$payment_intent = $this->mock_payment_intent();
+
+		$charge = $this->mock_charge( $order->get_id() );
+
+		$settings = $this->mock_settings();
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_intent' )
+			->with( '42' )
+			->willReturn( $payment_intent );
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'get_charge' )
+			->with( $payment_intent->get_charge_id() )
+			->willReturn( $charge );
+
+		$this->mock_wcpay_gateway
+			->expects( $this->exactly( 4 ) )
+			->method( 'get_option' )
+			->willReturnOnConsecutiveCalls( $settings['business_name'], $settings['support_info']['address'], $settings['support_info']['phone'], $settings['support_info']['email'] );
+
+		$this->mock_receipts_service
+			->expects( $this->once() )
+			->method( 'get_receipt_markup' )
+			->with( $settings, $this->isInstanceOf( WC_Order::class ), $charge )
+			->willThrowException( new Exception( 'Something bad' ) );
+
+		$request = new WP_REST_Request( 'GET' );
+		$request->set_param( 'payment_id', 42 );
+
+		$response = $this->controller->generate_print_receipt( $request );
+
+		$this->assertInstanceOf( 'WP_Error', $response );
+		$data = $response->get_error_data();
+		$this->assertArrayHasKey( 'status', $data );
+		$this->assertEquals( 500, $data['status'] );
```

Comment:
Please avoid `assertEquals` in tests, `assertSame` does the same + verifying types.